pkg/kvstore/backend.go

```diff
@@ -47,8 +51,25 @@ func (d *dsBackend) Save(ctx context.Context, r io.Reader, path string) error {
 	if err := d.ds.Upload(ctx, r, tmpfile); err != nil {
 		return err
 	}
+	log.Debugf("kv store upload of file (%s) was successful", tmpfile)
 
 	if err := d.ds.Mv(ctx, tmpfile, path); err != nil {
+		log.Debugf("failed to move file during kv store save operation at path (%s) to path (%s) with error (%s)", tmpfile, path, err.Error())
+
+		// The below case can occur on VSAN in slow environments(especially nested).
+		// we will reattempt the move since it might take some time for the upload to replicate before presenting.
+		if isFileFault(err) {
+			log.Debugf("FileNotFound fault returned during a kv store move operation. retrying move operation. tmpfile (%s), path (%s)", tmpfile, path)
+
+			moveOperation := func() error {
```

Comment:
can we wrap the retry logic from line 56? I mean might not need this additional error check outside of retry